### PR TITLE
create new end point for accessing styled outputs by name

### PR DESCRIPTION
**Pull request by MikeVanMourik**
# Describe your changes

because we are going to change the styled output table to use auto increment for its ids we added the function to use the name instead to find styled outputs

## Type of change

Please check only ONE option.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## How was this tested?

ran locally with postman

# Checklist before requesting a review
- [x] I have reviewed and tested my changes
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I selected `develop` as the base branch and not `main`, or the pull request is a hotfix that needs to be done directly on `main`
- [x] I double checked all my changes and they contain no temporary test code, no code that is commented out and no changes that are not part of this branch
- [x] I added new unit tests for my changes if applicable
note: i made a task to remind me to update the documentation once this is in

# Related pull requests

# Link to Asana ticket

Jira

### CODE REVIEW

File: Api/Modules/Queries/Controllers/StyledOutputController.cs [hunk 1]
```diff
@@ -39,13 +38,35 @@ public StyledOutputController(IStyledOutputService styledOutputService)
...
+        /// <summary>
+        /// Gets the queries that can be used for an export in the export module by name instead of id.
+        /// </summary>
+        /// <returns></returns>
+        [HttpPost]
+        [Route("api/v3/[controller]-named/{name:required}")]
+        [ProducesResponseType(StatusCodes.Status200OK)]
+        [ProducesResponseType(StatusCodes.Status401Unauthorized)]
+        [ProducesResponseType(StatusCodes.Status404NotFound)]
+        public async Task<IActionResult> GetStyledOutputResultJson(string name, [FromBody] List<KeyValuePair<string, object>> parameters, bool stripNewlinesAndTabs = false,[FromQuery] int resultsPerPage = 500, [FromQuery] int page = 0)
```
Comment (HansHappyHorizon) [inline Api/Modules/Queries/Controllers/StyledOutputController.cs:59]: This method is async, so the name should be `GetStyledOutputResultJsonAsync`. Please update the name.

File: Api/Modules/Queries/Interfaces/IStyledOutputService.cs [hunk 2]
```diff
@@ -11,6 +11,14 @@ namespace Api.Modules.Queries.Interfaces
     /// </summary>
     public interface IStyledOutputService
     {
+        /// <summary>
+        /// Find a styled output id based on its name.
+        /// </summary>
+        /// <param name="name">The name from wiser_styled_output.</param>
+        /// <returns>The results the id of the found styled output or instead returns -1 when not found .</returns>
+        ///
```
Comment (HansHappyHorizon) [inline Api/Modules/Queries/Interfaces/IStyledOutputService.cs:19]: Please remove this empty comment line.

File: Api/Modules/Queries/Services/StyledOutputService.cs [hunk 3]
```diff
@@ -74,6 +74,21 @@ public StyledOutputService(IOptions<StyledOutputSettings> apiSettings, IDatabase
...
+        {
+            var formatQuery = $"SELECT id FROM `wiser_styled_output` WHERE `name` = {name.ToMySqlSafeValue(true)} LIMIT 1";
+            
+            var dataTable = await clientDatabaseConnection.GetAsync(formatQuery);
+            
+            if (dataTable.Rows.Count == 0)
+            {
+                return -1;
+            }
+n
```
Comment (HansHappyHorizon) [inline Api/Modules/Queries/Services/StyledOutputService.cs:89]: I don't think this will compile, seeing as there's a random `n` here.

**Review by HansHappyHorizon — CHANGES_REQUESTED** [review]
Found some issues with the code. Please fix these.

File: Api/Modules/Queries/Services/StyledOutputService.cs (hunk 3 above)
Comment (MikeVanMourik) [reply Api/Modules/Queries/Services/StyledOutputService.cs:89]: correct, im guessing i hit my keyboard by accident after testing a compile, didnt notice it on the commit

File: Api/Modules/Queries/Interfaces/IStyledOutputService.cs (hunk 2 above)
Comment (MikeVanMourik) [reply Api/Modules/Queries/Interfaces/IStyledOutputService.cs:19]: done

File: Api/Modules/Queries/Controllers/StyledOutputController.cs (hunk 1 above)
Comment (MikeVanMourik) [reply Api/Modules/Queries/Controllers/StyledOutputController.cs:59]: alrighty

**Review by HansHappyHorizon — APPROVED** [review]

**Pull request merged**